### PR TITLE
desktop: Fix rendering when window is has unexpected size (close #1631)

### LINKED ISSUE #1631: Very tall flashes only render as pure white in the Windows Desktop version
Example flash: https://www.homestuck.com/flash/hs2/05492/05492.swf

This flash works perfectly in the web player. However, when opening it using ruffle.exe, nothing renders and you can only see the white background. You can still hear that the flash is working correctly based on the background sound effects.

Testing ruffle.exe nightly version 2020-11-20 on Windows 10.

### PR DESCRIPTION
**Pull request by MrCheeze**
Winit's WindowBuilder will sometimes create a window whose size is
smaller than requested. Therefore, instead of assuming the actual
window size is requested size * scaling, just check what the true
size is. (window.inner_size() also takes scaling into account.)

This fixes at least one issue where rendering would fail on Windows
with the Vulkan backend due to the incorrect size being used.

~~On a side note: winit is apparently limiting the window size to the screen resolution. This pull request doesn't change that behaviour, because when [I tried to do so](https://github.com/MrCheeze/ruffle/commit/e4cae4f23974a9b22b31c93bd5da52171a6e4401) it ended up causing crashes in the DX11 backend for unclear reasons. But I think this change is valuable enough on its own.~~

### CODE REVIEW

**Comment by MrCheeze** [comment]
Added an extra change to the pull request so that the window size isn't limited to the screen resolution in the first place. This required bumping the wgpu-rs version slightly because the commit we were using til now had a crash whenever the window is resized in DX11.

File: desktop/src/main.rs [hunk 1]
```diff
@@ -197,9 +197,10 @@ fn run_player(opt: Opt) -> Result<(), Box<dyn std::error::Error>> {
         WindowBuilder::new()
             .with_title(format!("Ruffle - {}", window_title))
             .with_window_icon(Some(icon))
-            .with_inner_size(movie_size)
+            .with_max_inner_size(LogicalSize::new(10000, 10000))
```
Comment (Herschel) [inline desktop/src/main.rs:200]: Slightly-less-arbitrary `u32::MAX` might be a little better than arbitrary `10000`

Comment (MrCheeze) [reply desktop/src/main.rs]: Splitting the difference and updating it to i16::MAX since no flash is larger than that anyway.

**Comment by Herschel** [comment]
Thank you!

**Pull request merged**